src/pages/index.js

```diff
@@ -0,0 +1,274 @@
+                            </Text>
+                        </Box>
+                        <Settings
+                            searchRadius={parseInt(searchRadius)}
+                            showAllClubs={showAllClubs}
+                            useImperialSystem={useImperialSystem}
+                            style={{ marginRight: theme.space[2] }}
+                            onRadiusChange={e => {
+                                e.persist();
+                                this.setState({ searchRadius: e.target.value });
+                                this.onRadiusChange(e);
+                            }}
+                            onSystemChange={this.onSystemChange}
+                            onViewChange={this.onViewChange}
+                        />
+                    </Flex>
+                    <Flex justify={hasNoResults ? "center" : "initial"} py={4} style={{ marginLeft: -theme.space[2], marginTop: -theme.space[2] }} wrap>
+                        {
+                            filteredClubs.map(club => (
+                                <LazyLoad key={club.id} height={0} offset={100} once overflow>
+                                    <ClubCard
+                                        data={club}
+                                        distance={!showAllClubs && geolib.getDistance({ latitude: searchLat, longitude: searchLng }, club)}
+                                        showAllClubs={showAllClubs}
+                                        useImperialSystem={useImperialSystem}
+                                    />
+                                </LazyLoad>
+                            ))
+                        }
+                        {hasNoResults && <NoClubsFound />}
+                    </Flex>
+                </Container>
+                <Footer />
+            </Fragment>
+        );
+    }
+
+    onRadiusChange(e) {
+        const nextSearchRadius = parseInt(e.target.value);
+        const { clubs, searchLat, searchLng, useImperialSystem } = this.state;
+        this.setState({ searchRadius: nextSearchRadius });
+        this.setParams({ r: nextSearchRadius });
+        if(searchLat && searchLng) {
+            const filteredClubs = this.getFilteredClubs(clubs, {
+                searchLat: searchLat,
+                searchLng: searchLng,
+                searchRadius: nextSearchRadius,
+                useImperialSystem
+            });
+            this.setState({ filteredClubs });
+        }
+    }
+
+    async onSearchChange(e) {
+        const { mapsApiKey } = this.props.data.site.siteMetadata;
+        const {
+            clubs,
+            searchRadius,
+            searchValue,
+            showAllClubs,
+            useImperialSystem
+        } = this.state;
+        const hasSearchValue = searchValue.trim().length !== 0;
+        if(!hasSearchValue) {
```

Comment:
can be one line, no need for block

Reply:
Can't really tell from the highlighted code, but are you referring to `this.state`?